theatre/studio/src/StudioStore/StudioStore.ts

```diff
@@ -131,56 +134,77 @@ export default class StudioStore {
-                      {
-                        ...propAddress,
-                        trackId,
-                        position: seq.position,
-                        value: v as $FixMe,
-                        snappingFunction: seq.closestGridPosition,
-                      },
-                    )
-                  } else {
-                    stateEditors.coreByProject.historic.sheetsById.staticOverrides.byObject.setValueOfPrimitiveProp(
-                      {...propAddress, value: v},
-                    )
-                  }
-                },
-                getPointerParts(pointer as Pointer<$IntentionalAny>).path,
-              )
+              const propConfig = getPropConfigByPath(
+                root.template.config,
+                path,
+              ) as PropTypeConfig
+
+              const setStaticOrKeyframeProp = <T>(
+                value: T,
+                path: PathToProp,
+              ) => {
+                if (typeof value === 'undefined' || value === null) {
+                  return
+                }
+
+                const propAddress = {...root.address, pathToProp: path}
+
+                const trackId = get(sequenceTracksTree, path) as $FixMe as
+                  | SequenceTrackId
+                  | undefined
+                if (typeof trackId === 'string') {
+                  const propConfig = getPropConfigByPath(
+                    root.template.config,
+                    path,
+                  ) as PropTypeConfig | undefined
+                  // TODO: Make sure this causes no problems wrt decorated
+                  // or otherwise unserializable stuff that sanitize might return.
+                  // value needs to be serializable.
+                  if (propConfig?.sanitize) value = propConfig.sanitize(value)
+
+                  const seq = root.sheet.getSequence()
+                  seq.position = seq.closestGridPosition(seq.position)
+                  stateEditors.coreByProject.historic.sheetsById.sequence.setKeyframeAtPosition(
+                    {
+                      ...propAddress,
+                      trackId,
+                      position: seq.position,
+                      value: value as $FixMe,
+                      snappingFunction: seq.closestGridPosition,
+                    },
+                  )
+                } else if (propConfig !== undefined) {
+                  stateEditors.coreByProject.historic.sheetsById.staticOverrides.byObject.setValueOfPrimitiveProp(
+                    {...propAddress, value: value as $FixMe},
+                  )
+                }
+              }
+
+              // If we are dealing with a compound prop, we recurse through its
+              // nested properties. The root prop with path.length === 0 is
+              // implicitly considered a compound prop.
+              if (path.length === 0 || propConfig?.type === 'compound') {
```

Comment:
So, if propConfig is undefined, we shouldn't set that prop anyway, I think. Also, checking for `path.length === 0` looks brittle, because then `propConfig` _would_ be a compound.

Reply:
Oh, good catch, I was meaning remove the `?` because `propConfig` is never actually undefined, that was a remnant from some previous buggy code. `path.length === 0` is necessary because the root prop is an "implicit" compound prop, meaning it doesn't have `propConfig.type === 'compound'`, but should be treated as one.